mongo/database.go

```diff
@@ -161,8 +165,13 @@ func (db *Database) processRunCommand(ctx context.Context, cmd interface{},
 		Database(db.name).Deployment(db.client.deployment).ReadConcern(db.readConcern).Crypt(db.client.crypt), sess, nil
 }
 
-// RunCommand runs a command on the database. A user can supply a custom
-// context to this method, or nil to default to context.Background().
+// RunCommand executes the given command against the database.
+//
+// The runCommand parameter should be a document for the command to be executed. It cannot be nil.
+// This must be an order-preserving type such as bson.D. Map types such as bson.M are not valid.
+// If the command document contains a session ID or any transaction-specific fields, the behavior is undefined.
+//
+// The opts parameter can be used to specify options for this operation (see the options.RunCmdOptions documentation).
```

Comment:
Additionally, I would explain what the SingleResult represents and whether it will be in an error state if the command errors, and whether one can retrieve the command result document in the case of error.

Reply:
Because `SingleResult` is returned by other CRUD functions too, I think we should defer documenting this to the `SingleResult` documentation itself. The documentation for `SingleResult.DecodeBytes` already says "If there was an error from the operation that created this SingleResult then the error will be returned along with the result."